kernel/cpu/src/arch/x86_64/intel.rs

```diff
@@ -0,0 +1,81 @@
+use crate::IdleState;
+
+pub(crate) enum Model {
+    Broadwell,
+}
+
+impl Model {
+    pub(crate) fn current() -> Option<Self> {
+        let eax = unsafe { core::arch::x86_64::__cpuid(1).eax };
+        let model = (eax >> 4) & 0xf;
+        let extended_model = (eax >> 16) & 0xf;
```

Comment:
Does the raw_cpuid crate cover this? If so, we should use it; if not, we should add it there.